relay_messages_test.go

```diff
@@ -168,45 +182,34 @@ const (
 	resTotalCombinations
 )
 
-func (cr testCallRes) res() lazyCallRes {
-	f := NewFrame(100)
-	fh := FrameHeader{
-		size:        uint16(0xFF34),
-		messageType: messageTypeCallRes,
-		ID:          0xDEADBEEF,
+func (cr testCallRes) res(tb testing.TB) lazyCallRes {
+	params := testCallResParams{
+		payloadSize: 100,
 	}
-	f.Header = fh
-	fh.write(typed.NewWriteBuffer(f.headerBuffer))
-
-	payload := typed.NewWriteBuffer(f.Payload)
-
-	if cr&resIsContinued == 0 {
-		payload.WriteSingleByte(0) // flags
-	} else {
-		payload.WriteSingleByte(hasMoreFragmentsFlag) // flags
+	if cr&resIsContinued != 0 {
+		params.flags |= hasMoreFragmentsFlag
 	}
-
 	if cr&resIsOK == 0 {
-		payload.WriteSingleByte(1) // code not ok
-	} else {
-		payload.WriteSingleByte(0) // code ok
+		params.code = 1
 	}
-
-	headers := make(map[string]string)
 	if cr&resHasHeaders != 0 {
-		addRandomHeaders(headers)
+		kvMap := map[string]string{
+			"k1":      "v1",
+			"k222222": "",
+			"k3":      "thisisalonglongkey",
+		}
+		params.headers = append(params.headers, byte(len(kvMap)))
+		for k, v := range kvMap {
+			params.headers = append(params.headers, byte(len(k)))
+			params.headers = append(params.headers, []byte(k)...)
+			params.headers = append(params.headers, byte(len(v)))
+			params.headers = append(params.headers, []byte(v)...)
+		}
```

Comment:
Unlike the call-req methods, I think I understand the separation here:
 * `testCallRes.res(t)` allows building a frame with some "predefined" values based on the bits set
 * `newCallResFrame` allows building a call res frame with a specific set of parameters

The first uses the latter, and the latter is responsible for all the frame encoding. Given that, should `headers` be a `map[string]string`? that way all the encoding logic is in a single place